src/lm.jl

```diff
@@ -156,7 +156,30 @@ function coeftable(mm::LinearModel)
               ["x$i" for i = 1:size(mm.pp.X, 2)], 4)
 end
 
-predict(mm::LinearModel, newx::Matrix) =  newx * coef(mm)
+# Different methods of prediction for the different factorization types
+
+predict(mm::LinearModel, newx::AbstractMatrix) = newx * coef(mm)
+
+function predict(mm::LinearModel, newx::AbstractMatrix, interval_type::Symbol, level = 0.95)
+    prediction = newx * coef(mm)
+    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented
+
+    R = cholfact!(mm.pp)[:U] #get the R matrix from the QR factorization
+    residvar = (ones(3,1) * deviance(mm)/dof_residual(mm))'
+
+    function _local(x::StridedVector)
+        Ac_ldiv_B!(R,x)
+        residvar * (x⋅x)
+    end
+
+    interval = similar(prediction)
+    for i in 1:length(interval)
+        interval[i] = _local(newx[i,:])[1]
+    end
+
+    interval = quantile(TDist(dof_residual(mm)), (1 - level)/2) * sqrt(interval)
```

Comment:
I don't think I am doing it right (`predict_old` is the current, `predict` my attempt at your suggestion):
```julia
predict(mm::LinearModel, newx::AbstractMatrix) = newx * coef(mm)

function predict(mm::LinearModel, x::StridedVector, tvalue, residvar, R, interval_type)
    Ac_ldiv_B!(R,x)
    tmp = residvar * (x⋅x) # ready for modifying if I need a prediction interval
    tvalue * sqrt(tmp)[1]
end

function predict(mm::LinearModel, newx::AbstractMatrix, interval_type::Symbol, level = 0.95)
    prediction = newx * coef(mm)
    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented

    R = cholfact!(mm.pp)[:U] #get the R matrix from the QR factorization
    residvar = (ones(size(newx,2),1) * deviance(mm)/dof_residual(mm))'
    tvalue = quantile(TDist(dof_residual(mm)), (1 - level)/2)
    interval = similar(prediction)

    for i in 1:length(interval)
        interval[i] = predict(mm, newx[i,:], tvalue, residvar, R, interval_type)
    end

    prediction, prediction .+ interval, prediction .- interval
end

function predict_old(mm::LinearModel, newx::AbstractMatrix, interval_type::Symbol, level = 0.95)
    prediction = newx * coef(mm)
    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented

    R = cholfact!(mm.pp)[:U] #get the R matrix from the QR factorization
    residvar = (ones(size(newx,2),1) * deviance(mm)/dof_residual(mm))'


    function _local(x::StridedVector)
        Ac_ldiv_B!(R,x)
        residvar * (x⋅x)
    end

    interval2 = similar(prediction)
    for i in 1:length(interval2)
        interval2[i] = _local(newx[i,:])[1]
    end

    interval2 = quantile(TDist(dof_residual(mm)), (1 - level)/2) * sqrt(interval2)
    prediction, prediction .+ interval2, prediction .- interval2
end
```
![skaermbillede 2016-12-16 kl 13 39 00](https://cloud.githubusercontent.com/assets/8429802/21263092/105a8be6-c395-11e6-9e38-f0fbccf7cae8.png)